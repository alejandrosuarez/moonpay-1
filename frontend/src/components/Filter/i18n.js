const translations = {
  fr: {
    titleSort: "Trier",
    titleSortName: "Trier par nom",
    titleSortCode: "Trier par code",
    titleSortRandom: "Mélanger aléatoirement",
    labelSupportedUS: "Fonctionne aux US",
    labelSupportsTestMode: "Supports en Mode Test",
  },
  en: {
    titleSort: "Sort",
    titleSortName: "Sort by name",
    titleSortCode: "Sort by code",
    titleSortRandom: "Random shuffle",
    labelSupportedUS: "Supported in US",
    labelSupportsTestMode: "Supports in Test Mode",
  },
};

export default translations;
