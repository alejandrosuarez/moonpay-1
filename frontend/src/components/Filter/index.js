import React from "react";
import "./index.scss";
import translations from "./i18n";
import {
  SORT_BY_CODE,
  SORT_BY_NAME,
  SORT_BY_RANDOM,
} from "../../settings/constants";
import { CheckBox, Box, Menu, Text } from "grommet";
import { PowerCycle, FormDown, Descend } from "grommet-icons";
import { useDispatch, useSelector } from "react-redux";
import {
  filterBySupportedInUS,
  filterBySupportsInTestMode,
  filterSort,
} from "../../redux/filter/filter.actions";

const Filter = () => {
  // Selector to get the redux state
  const filter = useSelector((state) => state.filter);
  const { locale } = useSelector((state) => state.locale);

  // Dispatch to update the redux state
  const dispatch = useDispatch();

  return (
    <Box
      pad="medium"
      direction="column"
      margin={{ vertical: "medium" }}
      className="p-filter"
      gap="medium"
    >
      <CheckBox
        toggle
        checked={filter.isSupportedInUS}
        onChange={(event) =>
          dispatch(filterBySupportedInUS(event.target.checked))
        }
        label={translations[locale].labelSupportedUS}
      />
      <CheckBox
        toggle
        checked={filter.isSupportsTestMode}
        onChange={(event) =>
          dispatch(filterBySupportsInTestMode(event.target.checked))
        }
        label={translations[locale].labelSupportsTestMode}
      />
      <Menu
        plain
        items={[
          {
            label: (
              <Box alignSelf="center">
                <Text color={filter.sortBy === SORT_BY_NAME ? "accent-1" : ""}>
                  {translations[locale].titleSortName}
                </Text>
              </Box>
            ),
            onClick: () => {
              dispatch(filterSort(SORT_BY_NAME));
            },
            icon: (
              <Box pad="small">
                <Descend size="medium" />
              </Box>
            ),
          },
          {
            label: (
              <Box alignSelf="center">
                <Text color={filter.sortBy === SORT_BY_CODE ? "accent-1" : ""}>
                  {translations[locale].titleSortCode}
                </Text>
              </Box>
            ),
            onClick: () => {
              dispatch(filterSort(SORT_BY_CODE));
            },
            icon: (
              <Box pad="small">
                <Descend size="medium" />
              </Box>
            ),
          },
          {
            label: (
              <Box alignSelf="center">
                <Text
                  color={filter.sortBy === SORT_BY_RANDOM ? "accent-1" : ""}
                >
                  {translations[locale].titleSortRandom}
                </Text>
              </Box>
            ),
            onClick: () => {
              dispatch(filterSort(SORT_BY_RANDOM));
            },
            icon: (
              <Box pad="small">
                <PowerCycle size="medium" />
              </Box>
            ),
          },
        ]}
      >
        {() => (
          <Box direction="row" gap="small" pad="small">
            <FormDown />
            <Text>{translations[locale].titleSort}</Text>
          </Box>
        )}
      </Menu>
    </Box>
  );
};

export default Filter;
