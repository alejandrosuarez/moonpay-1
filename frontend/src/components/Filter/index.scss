@import "../../styles/index.scss";

.p-filter {
  border-radius: $border-radius;
  box-shadow: 0 0 rem(10) rem(-5) $color-grey;
}
