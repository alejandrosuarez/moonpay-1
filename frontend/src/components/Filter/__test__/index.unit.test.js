import React from "react";
import ReactDom from "react-dom";
import Filter from "../index";
import { cleanup } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import { create } from "react-test-renderer";

jest.mock("react-redux", () => ({
  useSelector: () => ({ locale: "en" }),
  useDispatch: () => jest.fn(),
}));

describe("Filter Component", () => {
  afterEach(cleanup);

  it("Renders without crashing", () => {
    const div = document.createElement("div");
    ReactDom.render(<Filter />, div);
  });

  it("Matches snapshot empty", () => {
    const currency = create(<Filter />).toJSON();
    expect(currency).toMatchSnapshot();
  });
});
