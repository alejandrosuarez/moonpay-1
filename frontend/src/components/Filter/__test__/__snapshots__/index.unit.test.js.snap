// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Matches snapshot empty 1`] = `
<div
  className="StyledBox-sc-13pk1d4-0 gCNHBc p-filter"
>
  <label
    checked={false}
    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 htOOow"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="StyledBox-sc-13pk1d4-0 jbIyUK StyledCheckBox-sc-1dbk5ju-6 hOFfoi"
    >
      <input
        checked={false}
        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 fYuA-Ds"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <span
        checked={false}
        className="StyledCheckBox__StyledCheckBoxToggle-sc-1dbk5ju-4 imINeg"
      >
        <span
          checked={false}
          className="StyledCheckBox__StyledCheckBoxKnob-sc-1dbk5ju-5 dVITJo"
        />
      </span>
    </div>
    <span>
      Supported in US
    </span>
  </label>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDNLiy"
  />
  <label
    checked={false}
    className="StyledCheckBox__StyledCheckBoxContainer-sc-1dbk5ju-1 htOOow"
    onClick={[Function]}
  >
    <div
      checked={false}
      className="StyledBox-sc-13pk1d4-0 jbIyUK StyledCheckBox-sc-1dbk5ju-6 hOFfoi"
    >
      <input
        checked={false}
        className="StyledCheckBox__StyledCheckBoxInput-sc-1dbk5ju-2 fYuA-Ds"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        type="checkbox"
      />
      <span
        checked={false}
        className="StyledCheckBox__StyledCheckBoxToggle-sc-1dbk5ju-4 imINeg"
      >
        <span
          checked={false}
          className="StyledCheckBox__StyledCheckBoxKnob-sc-1dbk5ju-5 dVITJo"
        />
      </span>
    </div>
    <span>
      Supports in Test Mode
    </span>
  </label>
  <div
    className="StyledBox__StyledBoxGap-sc-13pk1d4-1 kDNLiy"
  />
  <button
    aria-label="Open Menu"
    className="StyledButton-sc-323bzc-0 ebqXLG"
    onBlur={[Function]}
    onClick={[Function]}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    type="button"
  >
    <div
      className="StyledBox-sc-13pk1d4-0 htdGLi"
    >
      <svg
        aria-label="FormDown"
        className="StyledIcon-ofa7kd-0 dEeaTZ"
        viewBox="0 0 24 24"
      >
        <polyline
          fill="none"
          points="18 9 12 15 6 9"
          stroke="#000"
          strokeWidth="2"
        />
      </svg>
      <div
        className="StyledBox__StyledBoxGap-sc-13pk1d4-1 eBVJAG"
      />
      <span
        className="StyledText-sc-1sadyjn-0 DkZvS"
      >
        Sort
      </span>
    </div>
  </button>
</div>
`;
